## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 492814`
Commit: `d73a3b2875aff836d2788121566a9119ef7a0b86`

---
### `x86_64-linux`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>envoy</li>
    <li>perf_data_converter</li>
    <li>pomerium</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 33 packages built:</summary>
  <ul>
    <li>anytype</li>
    <li>anytype-heart</li>
    <li>bant</li>
    <li>fcitx5-mozc</li>
    <li>fcitx5-mozc-ut</li>
    <li>mozc (ibus-engines.mozc)</li>
    <li>mozc-ut (ibus-engines.mozc-ut)</li>
    <li>protoc-gen-js</li>
    <li>python313Packages.arviz</li>
    <li>python313Packages.arviz.dist</li>
    <li>python313Packages.bambi</li>
    <li>python313Packages.bambi.dist</li>
    <li>python313Packages.corner</li>
    <li>python313Packages.corner.dist</li>
    <li>python313Packages.distrax</li>
    <li>python313Packages.distrax.dist</li>
    <li>python313Packages.flowjax</li>
    <li>python313Packages.flowjax.dist</li>
    <li>python313Packages.numpyro</li>
    <li>python313Packages.numpyro.dist</li>
    <li>python313Packages.nutpie</li>
    <li>python313Packages.nutpie.dist</li>
    <li>python313Packages.pymc</li>
    <li>python313Packages.pymc.dist</li>
    <li>python313Packages.pytensor</li>
    <li>python313Packages.pytensor.dist</li>
    <li>python313Packages.rlax</li>
    <li>python313Packages.rlax.dist</li>
    <li>python313Packages.tensorflow-probability</li>
    <li>python313Packages.tensorflow-probability.dist</li>
    <li>verible</li>
    <li>veridian</li>
    <li>xla</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>perf_data_converter</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 34 packages built:</summary>
  <ul>
    <li>anytype</li>
    <li>anytype-heart</li>
    <li>bant</li>
    <li>envoy</li>
    <li>fcitx5-mozc</li>
    <li>fcitx5-mozc-ut</li>
    <li>mozc (ibus-engines.mozc)</li>
    <li>mozc-ut (ibus-engines.mozc-ut)</li>
    <li>pomerium</li>
    <li>protoc-gen-js</li>
    <li>python313Packages.arviz</li>
    <li>python313Packages.arviz.dist</li>
    <li>python313Packages.bambi</li>
    <li>python313Packages.bambi.dist</li>
    <li>python313Packages.corner</li>
    <li>python313Packages.corner.dist</li>
    <li>python313Packages.distrax</li>
    <li>python313Packages.distrax.dist</li>
    <li>python313Packages.flowjax</li>
    <li>python313Packages.flowjax.dist</li>
    <li>python313Packages.numpyro</li>
    <li>python313Packages.numpyro.dist</li>
    <li>python313Packages.nutpie</li>
    <li>python313Packages.nutpie.dist</li>
    <li>python313Packages.pymc</li>
    <li>python313Packages.pymc.dist</li>
    <li>python313Packages.pytensor</li>
    <li>python313Packages.pytensor.dist</li>
    <li>python313Packages.rlax</li>
    <li>python313Packages.rlax.dist</li>
    <li>python313Packages.tensorflow-probability</li>
    <li>python313Packages.tensorflow-probability.dist</li>
    <li>verible</li>
    <li>veridian</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>protoc-gen-js</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>protoc-gen-js</li>
    <li>verible</li>
    <li>xla</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>perf_data_converter</summary>
<pre>                 from external/abseil-cpp~/absl/flags/reflection.h:29:
external/abseil-cpp~/absl/container/internal/container_memory.h: In function &#x27;void* absl::lts_20240116::container_internal::Allocate(Alloc*, size_t)&#x27;:
external/abseil-cpp~/absl/container/internal/container_memory.h:66:27: error: &#x27;uintptr_t&#x27; does not name a type [-Wtemplate-body]
   66 |   assert(reinterpret_cast&lt;uintptr_t&gt;(p) % Alignment == 0 &amp;&amp;
      |                           ^~~~~~~~~
external/abseil-cpp~/absl/container/internal/container_memory.h:31:1: note: &#x27;uintptr_t&#x27; is defined in header &#x27;&lt;cstdint&gt;&#x27;; this is probably fixable by adding &#x27;#include &lt;cstdint&gt;&#x27;
   30 | #include &quot;absl/utility/utility.h&quot;
  +++ |+#include &lt;cstdint&gt;
   31 | 
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 4.917s, Critical Path: 2.54s
INFO: 838 processes: 599 internal, 239 linux-sandbox.
ERROR: Build did NOT complete successfully
//src:intervalmap_test                                                NO STATUS
//src:perf_data_converter_test                                        NO STATUS
//src:perf_to_profile_lib_test                                        NO STATUS
//src:perf_data_handler_test                                    FAILED TO BUILD

Executed 0 out of 4 tests: 1 fails to build and 3 were skipped.
</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>perf_data_converter</summary>
<pre>                 from external/abseil-cpp~/absl/flags/reflection.h:29:
external/abseil-cpp~/absl/container/internal/container_memory.h: In function &#x27;void* absl::lts_20240116::container_internal::Allocate(Alloc*, size_t)&#x27;:
external/abseil-cpp~/absl/container/internal/container_memory.h:66:27: error: &#x27;uintptr_t&#x27; does not name a type [-Wtemplate-body]
   66 |   assert(reinterpret_cast&lt;uintptr_t&gt;(p) % Alignment == 0 &amp;&amp;
      |                           ^~~~~~~~~
external/abseil-cpp~/absl/container/internal/container_memory.h:31:1: note: &#x27;uintptr_t&#x27; is defined in header &#x27;&lt;cstdint&gt;&#x27;; this is probably fixable by adding &#x27;#include &lt;cstdint&gt;&#x27;
   30 | #include &quot;absl/utility/utility.h&quot;
  +++ |+#include &lt;cstdint&gt;
   31 | 
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 32.363s, Critical Path: 11.33s
INFO: 1572 processes: 646 internal, 926 linux-sandbox.
ERROR: Build did NOT complete successfully
//src:perf_data_converter_test                                        NO STATUS
//src:perf_to_profile_lib_test                                        NO STATUS
//src:perf_data_handler_test                                    FAILED TO BUILD
//src:intervalmap_test                                                   PASSED in 0.0s

Executed 1 out of 4 tests: 1 test passes, 1 fails to build and 2 were skipped.
</pre>
</details>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>protoc-gen-js</summary>
<pre>In file included from external/com_google_absl/absl/base/nullability.h:153:
In file included from external/com_google_absl/absl/base/internal/nullability_impl.h:22:
external/com_google_absl/absl/meta/type_traits.h:511:36: warning: builtin __is_trivially_relocatable is deprecated; use __builtin_is_cpp_trivially_relocatable instead [-Wdeprecated-builtins]
  511 |     : std::integral_constant&lt;bool, __is_trivially_relocatable(T)&gt; {};
      |                                    ^
In file included from external/com_google_absl/absl/strings/escaping.cc:15:
In file included from external/com_google_absl/absl/strings/escaping.h:32:
In file included from external/com_google_absl/absl/strings/str_join.h:59:
In file included from external/com_google_absl/absl/strings/internal/str_join_internal.h:43:
In file included from external/com_google_absl/absl/strings/str_cat.h:108:
external/com_google_absl/absl/strings/numbers.h:27:10: error: module com_google_absl//absl/strings:strings does not depend on a module exporting &#x27;tmmintrin.h&#x27;
   27 | #include &lt;tmmintrin.h&gt;
      |          ^
1 warning and 1 error generated.
Target //generator:protoc-gen-js failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 40.244s, Critical Path: 1.62s
INFO: 1158 processes: 1027 internal, 131 processwrapper-sandbox.
ERROR: Build did NOT complete successfully
</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>protoc-gen-js</summary>
<pre>ERROR: /nix/var/nix/builds/nix-58768-3729406321/output/external/com_google_protobuf/third_party/utf8_range/BUILD.bazel:32:11: Compiling third_party/utf8_range/utf8_range.c failed: (Exit 1): cc_wrapper.sh failed: error executing CppCompile command (from target @@com_google_protobuf//third_party/utf8_range:utf8_range) external/local_config_cc/cc_wrapper.sh -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics ... (remaining 61 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from external/com_google_protobuf/third_party/utf8_range/utf8_range.c:18:
external/com_google_protobuf/third_party/utf8_range/utf8_range.h:4:10: error: module com_google_protobuf//third_party/utf8_range:utf8_range does not depend on a module exporting &#x27;stddef.h&#x27;
    4 | #include &lt;stddef.h&gt;
      |          ^
external/com_google_protobuf/third_party/utf8_range/utf8_range.c:20:10: error: module com_google_protobuf//third_party/utf8_range:utf8_range does not depend on a module exporting &#x27;stddef.h&#x27;
   20 | #include &lt;stddef.h&gt;
      |          ^
external/com_google_protobuf/third_party/utf8_range/utf8_range.c:21:10: error: module com_google_protobuf//third_party/utf8_range:utf8_range does not depend on a module exporting &#x27;stdint.h&#x27;
   21 | #include &lt;stdint.h&gt;
      |          ^
3 errors generated.
Target //generator:protoc-gen-js failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 5.286s, Critical Path: 1.11s
INFO: 838 processes: 832 internal, 6 processwrapper-sandbox.
ERROR: Build did NOT complete successfully
</pre>
</details>
<details>
<summary>verible</summary>
<pre>//verible/verilog/tools/lint:verilog-syntax-lexer-fail_test           NO STATUS
//verible/verilog/tools/lint:void-cast-allrules_test                  NO STATUS
//verible/verilog/tools/lint:void-cast-defaultrules_test              NO STATUS
//verible/verilog/tools/lint:void-cast-minusrule_test                 NO STATUS
//verible/verilog/tools/lint:void-cast-onerule_test                   NO STATUS
//verible/verilog/tools/lint:void-cast-syntax_test                    NO STATUS
//verible/verilog/tools/ls:autoexpand_test                            NO STATUS
//verible/verilog/tools/ls:lsp-parse-buffer_test                      NO STATUS
//verible/verilog/tools/ls:symbol-table-handler_test                  NO STATUS
//verible/verilog/tools/ls:verible-verilog-ls_test                    NO STATUS
//verible/verilog/tools/ls:verilog-language-server_test               NO STATUS
//verible/verilog/tools/obfuscator:obfuscate_test                     NO STATUS
//verible/verilog/tools/preprocessor:verilog-preprocessor_test        NO STATUS
//verible/verilog/tools/project:project-tool_test                     NO STATUS
//verible/verilog/tools/syntax:verilog-syntax_test                    NO STATUS
//verible/verilog/transform:obfuscate_test                            NO STATUS
//verible/verilog/transform:strip-comments_test                       NO STATUS

Executed 0 out of 608 tests: 608 were skipped.
</pre>
</details>
</details>

--- a/pkgs/by-name/xl/xla/package.nix
+++ b/pkgs/by-name/xl/xla/package.nix
@@ -1,6 +1,7 @@
 {
   bazel_7,
   buildBazelPackage,
+  cctools,
   curl,
   double-conversion,
   fetchFromGitHub,
@@ -18,6 +19,7 @@
   python3,
   snappy,
   sqlite,
+  stdenv,
   which,
   zlib,
 }:
@@ -71,7 +73,8 @@ in
     gitMinimal
     pythonEnv # necessary for some patchShebang'ing
     which
-  ];
+  ]
+  ++ lib.optional stdenv.hostPlatform.isDarwin cctools.libtool;
 
   buildInputs = [
     curl
@@ -136,6 +139,14 @@ in
     "--host_cxxopt=cstdint"
     # Exclude mobile/iOS targets that have Bazel incompatibilities
     "--build_tag_filters=-mobile,-ios"
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    # Set macOS deployment target to suppress availability errors (e.g.
+    # futimens requires 10.13+) in both target and host-tool builds.
+    "--copt=-mmacosx-version-min=11.0"
+    "--host_copt=-mmacosx-version-min=11.0"
+    "--linkopt=-mmacosx-version-min=11.0"
+    "--host_linkopt=-mmacosx-version-min=11.0"
   ];
 
   removeRulesCC = false;
@@ -146,10 +157,16 @@ in
   removeLocal = false;
 
   fetchAttrs = {
-    sha256 = "sha256-OJfSqDlEC+yhWwwMwaD5HGvuHm8OWk+yQxmbH0/gZ88=";
+    sha256 =
+      {
+        x86_64-linux = "sha256-OJfSqDlEC+yhWwwMwaD5HGvuHm8OWk+yQxmbH0/gZ88=";
+        aarch64-darwin = "sha256-sKxtdgozCV1on1gd+bm8FKyFxP0u70Hs24OdLNA3jh0=";
+      }
+      .${stdenv.hostPlatform.system} or (throw "unsupported system: ${stdenv.hostPlatform.system}");
     preInstall = ''
       rm -rf $bazelOut/external/{local_config_python,\@local_config_python.marker}
       rm -rf $bazelOut/external/{local_config_sh,\@local_config_sh.marker}
+      rm -rf $bazelOut/external/{local_config_xcode,\@local_config_xcode.marker}
       rm -rf $bazelOut/external/{local_execution_config_python,\@local_execution_config_python.marker}
       rm -rf $bazelOut/external/{local_jdk,\@local_jdk.marker}
     '';
@@ -187,18 +204,9 @@ in
     homepage = "https://github.com/openxla/xla";
     license = lib.licenses.asl20;
     maintainers = with lib.maintainers; [ samuela ];
-    platforms = lib.platforms.unix;
-    badPlatforms = [
-      # ERROR: /build/output/external/local_config_cc/BUILD: no such target
-      # '@@local_config_cc//:cc-compiler-k8': target 'cc-compiler-k8' not declared in package ''
-      # defined by /build/output/external/local_config_cc/BUILD
-      "aarch64-linux"
-
-      # Bazel fails to build on darwin:
-      # ERROR: no such package '@@bazel_tools~xcode_configure_extension~local_config_xcode//':
-      # BUILD file not found in directory '' of external repository @@bazel_tools~xcode_configure_extension~local_config_xcode.
-      # Add a BUILD file to a directory to mark it as a package.
-      lib.systems.inspect.patterns.isDarwin
+    platforms = [
+      "x86_64-linux"
+      "aarch64-darwin"
     ];
   };
 }

